lib/coreboot/coreboot_table.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2017, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <console.h>
+#include <coreboot/coreboot_table.h>
+#include <debug.h>
+#include <mmio.h>
+#include <string.h>
+
+/*
+ * Structures describing coreboot's in-memory descriptor tables. See
+ * <coreboot>/src/commonlib/include/commonlib/coreboot_tables.h for
+ * canonical implementation.
+ */
+
+struct cb_header {
```

Comment:
Sorry, I don't really understand what you mean by "anonymous struct" here. To me, an anonymous struct is a struct without a name, like this:

struct {
  uint32_t a;
  uint32_t b;
} myvariable;

But all the structs you pointed out here are normal structs with names? I just have a nested annonymous union in cb_entry below, I can change that if you want...